src/common/com/intellij/plugins/haxe/ide/refactoring/introduce/HaxeIntroduceHandler.java

```diff
@@ -365,17 +378,25 @@ protected void performIntroduceWithDialog(HaxeIntroduceOperation operation) {
 
   protected void performInplaceIntroduce(HaxeIntroduceOperation operation) {
     final PsiElement statement = performRefactoring(operation);
-    final HaxeComponent target = PsiTreeUtil.findChildOfType(statement, HaxeComponent.class);
+    final HaxeComponent target = statement instanceof HaxeComponent ? (HaxeComponent)statement : PsiTreeUtil.findChildOfType(statement, HaxeComponent.class);
     if (target == null) {
       return;
     }
+
+    final LinkedHashSet<String> names = new LinkedHashSet<>();
+    if (operation.getName() != null) {
+      names.add(operation.getName());
+    } else {
+      names.addAll(operation.getSuggestedNames());
+    }
```

Comment:
Fixing the above bug exposed this one -- we wouldn't put the proper (given) name in place; instead, we would add the shortest suggestion.